crates/compiler-core/src/bytecode.rs

```diff
@@ -549,263 +549,146 @@ pub type NameIdx = u32;
+    /// Performs `is` comparison, or `is not` if `invert` is 1.
+    IsOp(Arg<Invert>),
+    Jump(Arg<Label>),
+    /// Peek at the top of the stack, and jump if this value is false.
+    /// Otherwise, pop top of stack.
+    JumpIfFalseOrPop(Arg<Label>),
     /// Performs exception matching for except.
     /// Tests whether the STACK[-2] is an exception matching STACK[-1].
     /// Pops STACK[-1] and pushes the boolean result of the test.
     JumpIfNotExcMatch(Arg<Label>),
-    Jump {
-        target: Arg<Label>,
-    },
-    /// Pop the top of the stack, and jump if this value is true.
-    PopJumpIfTrue {
-        target: Arg<Label>,
-    },
-    /// Pop the top of the stack, and jump if this value is false.
-    PopJumpIfFalse {
-        target: Arg<Label>,
-    },
     /// Peek at the top of the stack, and jump if this value is true.
     /// Otherwise, pop top of stack.
-    JumpIfTrueOrPop {
-        target: Arg<Label>,
-    },
-    /// Peek at the top of the stack, and jump if this value is false.
-    /// Otherwise, pop top of stack.
-    JumpIfFalseOrPop {
-        target: Arg<Label>,
-    },
+    JumpIfTrueOrPop(Arg<Label>),
+    ListAppend(Arg<u32>),
+    LoadAttr(Arg<NameIdx>),
+    LoadBuildClass,
+    LoadClassDeref(Arg<NameIdx>),
+    LoadClosure(Arg<NameIdx>),
+    /// Holds an index into constants vec.
+    LoadConst(Arg<u32>),
+    LoadDeref(Arg<NameIdx>),
+    LoadFast(Arg<NameIdx>),
+    LoadGlobal(Arg<NameIdx>),
+    LoadMethod(Arg<NameIdx>),
+    LoadNameAny(Arg<NameIdx>),
     MakeFunction,
-    SetFunctionAttribute {
-        attr: Arg<MakeFunctionFlags>,
-    },
-    CallFunctionPositional {
-        nargs: Arg<u32>,
-    },
-    CallFunctionKeyword {
-        nargs: Arg<u32>,
-    },
-    CallFunctionEx {
-        has_kwargs: Arg<bool>,
-    },
-    LoadMethod {
-        idx: Arg<NameIdx>,
-    },
-    CallMethodPositional {
-        nargs: Arg<u32>,
-    },
-    CallMethodKeyword {
-        nargs: Arg<u32>,
-    },
-    CallMethodEx {
-        has_kwargs: Arg<bool>,
-    },
-    ForIter {
-        target: Arg<Label>,
-    },
-    ReturnValue,
-    ReturnConst {
-        idx: Arg<u32>,
-    },
-    YieldValue,
-    YieldFrom,
-
+    MapAdd(Arg<u32>),
+    MatchClass(Arg<u32>),
+    MatchKeys,
+    MatchMapping,
+    MatchSequence,
+    Nop,
+    Pop,
+    PopBlock,
+    PopException,
+    /// Pop the top of the stack, and jump if this value is false.
+    PopJumpIfFalse(Arg<Label>),
+    /// Pop the top of the stack, and jump if this value is true.
+    PopJumpIfTrue(Arg<Label>),
+    PrintExpr,
+    Raise(Arg<RaiseKind>),
     /// Resume execution (e.g., at function start, after yield, etc.)
-    Resume {
-        arg: Arg<u32>,
-    },
-
+    Resume(Arg<u32>),
+    ReturnConst(Arg<u32>),
+    ReturnValue,
+    Reverse(Arg<u32>),
+    SetAdd(Arg<u32>),
+    SetFunctionAttribute(Arg<MakeFunctionFlags>),
     SetupAnnotation,
-    SetupLoop,
-
+    SetupAsyncWith(Arg<Label>),
+    SetupExcept(Arg<Label>),
```

Comment:
how about at least keeping the name as comment?

e.g.

```suggestion
    SetupExcept(Arg<Label>),  // handler
```

Reply:
@youknowone I can revert that change if that's a deal breaker for you:)

Reply:
> I like the new enum idea on #6313. Not sure about Label. Because all the labels are actually label.
> 
> If the consistency matters, giving label for every instruction looks better to me. It feels like redundancy while writing code, but we read code more than writing code. Getting information from code is easier than checking document.

IMO consistency matters, I can make them consistent in the other way around by having each Oparg to be stored as a named field if that's desired.

> The situation is also a bit different to your example. It is more like:
> 
> ```rust
> enum Foo {
>     CallMethodPositional { nargs: String },
>     ReturnConst { idx: String },
>     ...
> }
> ```
>
> Maybe giving different types for them will also make sense though.

I think that it will, but that is out of scope for this PR, I'll do it in a separate one
> cc @coolreader18